src/pkg/cli/compose/context.go

```diff
@@ -407,7 +418,7 @@ func createArchive(ctx context.Context, root string, dockerfile string, contentT
 	}
 
 	doProgress := term.StdoutCanColor() && term.IsTerminal()
-	err := WalkContextFolder(root, dockerfile, func(path string, de os.DirEntry, slashPath string) error {
+	err := walkContextFolder(root, dockerfile, writeIgnoreFileYes, func(path string, de os.DirEntry, slashPath string) error {
```

Comment:
Any reason we can't put this logic here? At a glance, I don't see why it should be coupled to `WalkContextFolder`.

```suggestion
	if dockerignore == "" {
		// Generate a default .dockerignore file if none exists (to be included in the context)
		term.Warn("No .dockerignore file found; creating default .dockerignore")
		var err error
		dockerignore, err = writeDefaultIgnoreFile(root, dotdockerignore)
		if err != nil {
			return fmt.Errorf("failed to write default .dockerignore file: %w", err)
		}
	}

	err := walkContextFolder(root, dockerfile, func(path string, de os.DirEntry, slashPath string) error {
```

Reply:
Because here we don't know whether there is a `.dockerignore` or not. We'd have to call `getDockerIgnorePatterns` twice.